### PR TITLE
Add typing for Passwordless APIs

### PR DESCRIPTION
**Pull request by mattgd**
## Description
Add typing for Passwordless APIs.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/passwordless.py [hunk 1]
```diff
@@ -1,61 +1,79 @@
...
+                from WorkOS. Defaults to the default Redirect URI in the
+                WorkOS dashboard.
+            state (str): Optional parameter that the redirect
+                URI received from WorkOS will contain. The state parameter
+                can be used to encode arbitrary information to help
+                restore application state between redirects.
+            type (str): The type of Passwordless Session to
+                create. Currently, the only supported value is 'MagicLink'.
+            expires_in (int): The number of seconds the Passwordless Session should live before expiring.
+                This value must be between 900 (15 minutes) and 86400 (24 hours), inclusive.
```
Comment (mattgd) [inline workos/passwordless.py:55]: Changes like these definitely need documenting, but fortunately the developer will be able to take their existing dict and just spread it on to the method call. For example:

4.x usage:
```
session_options = {
    ...
}

create_session(session_options)
```

5.x usage:
```
session_options = {
    ...
}

create_session(**session_options)
```

Comment (tribble) [reply workos/passwordless.py:55]: ... so this was like the API function we decided should accept a `dict`?

**Review by tribble — APPROVED** [review]
yay

File: workos/passwordless.py (hunk 1 above)
Comment (mattgd) [reply workos/passwordless.py:55]: I was under the impression we were standardizing on separate parameters like this change. Reason being, once we're on 3.12+, with have [PEP 692](https://peps.python.org/pep-0692/), we can type kwargs with a TypedDict. Also fewer existing methods use this "dict" parameter pattern.

Either way we landed, I'm realizing we probably should've logged that somewhere. Let me know if you think I should swap this back to one dict parameter.

**Pull request merged**